vault/logical_system.go

```diff
@@ -2742,9 +2742,18 @@ func (b *SystemBackend) handleMonitor(ctx context.Context, req *logical.Request,
 
 	flusher.Flush()
 
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
 	// Stream logs until the connection is closed.
 	for {
 		select {
+		// Periodically check for the seal status and return if core gets
+		// marked as sealed.
+		case <-ticker.C:
+			if b.Core.Sealed() {
+				return resp, nil
```

Comment:
I wonder if we should return an error in here so that the client can be notified that the server terminated the request.

Reply:
Yeah, sounds reasonable to me.

Reply:
Doesn't quite work for some reason, it's swallowing the error. I am going to dig a bit further into this.

Reply:
It seems this is due to how chunked transfer encoding works and the we that we send over chunked data. Once we flush the data, that sends over the status code to the client so any errors afterwards wouldn't change the response's status code. Additionally, we don't process any errors on our end if the writer has been already written to, which is why the error is swallowed.